exchanges/bitfinex/bitfinex.go

```diff
@@ -748,86 +748,86 @@ func (b *Bitfinex) GetTicker(ctx context.Context, symbol string) (*Ticker, error
 	if len(response) > 10 {
 		var ok bool
 		if t.FlashReturnRate, ok = response[0].(float64); !ok {
-			return nil, errors.New("unable to type assert flashReturnRate")
+			return nil, common.GetTypeAssertError("float64", response[0], "Ticker.Data.FlashReturnRate")
 		}
 		if t.Bid, ok = response[1].(float64); !ok {
-			return nil, errors.New("unable to type assert bid")
+			return nil, common.GetTypeAssertError("float64", response[1], "Ticker.Data.Bid")
 		}
-		var bidPeriod float64
-		bidPeriod, ok = response[2].(float64)
+		var bidPeriodF float64
+		bidPeriodF, ok = response[2].(float64)
 		if !ok {
-			return nil, errors.New("unable to type assert bidPeriod")
+			return nil, common.GetTypeAssertError("float64", response[2], "Ticker.Data.BidPeriod")
 		}
-		t.BidPeriod = int64(bidPeriod)
+		t.BidPeriod = int64(bidPeriodF)
 		if t.BidSize, ok = response[3].(float64); !ok {
-			return nil, errors.New("unable to type assert bidSize")
+			return nil, common.GetTypeAssertError("float64", response[3], "Ticker.Data.BidSize")
```

Comment:
An unfortunate error I am receiving frequently enough is from the following data in the response:
```
["tALT2612:USD",100,null,100,null,null,null,null,null,null,null],["tALT2612:UST",100,null,100,null,null,null,null,null,null,null]
```

I'll have a think on what to do for it, since returning the whole function in error because bitfinex didnt populate it properly seems annoying

Reply:
I actually had the same problem, and thought about it the same.
I ended up thinking "Meh. Maybe this is the most representative error to throw, really"
Otherwise `common.ErrAPIThrowingShitAgainKThx`

But seriously; Two of us had the same thought here. I'll come up with an error that's more representative than pretending like one field is the problem, and a way to categorise that.

Reply:
Fixed fd76eda6

In essence what I've done to address your problem:
* Improved error clarity (erm. Hopefully)
* Process all tickers and collect all errors
* Add specific tests for this eventuality
* Exclude response format errors from GetTickerBatch, whilst allowing it to enforce two know tickers are there